Sponsor benefit provided assets

ewdurbin opened the pull request:
WIP, for initial review by @berinhard. Creates the ability to configure "Provided Assets" associated with a sponsorship that will be fulfilled by the PSF.

The initial "ProvidedTextAsset" is intended to be used for PyCon US 2022 voucher codes, which will be unique to each voucher "type" and sponsorship level.

Additionally, we will likely include "ProvidedFileAsset" that will be used for all sponsorships with Expo Hall benefits to share with them a common "Exhibitor Packet"

ewdurbin committed:
add missing migration

ewdurbin committed:
WIP: "Provided Assets" from PSF to sponsors

creates the ability to configure "Provided Assets" associated with a sponsorship that will be fulfilled by the PSF.

The initial "ProvidedTextAsset" is intended to be used for PyCon US 2022 voucher codes, which will be unique to each voucher "type" and sponsorship level.

Additionally, we will likely include "ProvidedFileAsset" that will be used for all sponsorships with Expo Hall benefits to share with them a common "Exhibitor Packet"

ewdurbin committed:
upload files/shared provided file assets for configured benefits

berinhard reviewed (commented):
@ewdurbin so far the code seems fine to me. I tested it locally and admin seems to be working as expected. I tested the process locally and the user sponsorship dashboard got updated with the provided assets. But the form are the next steps, right?

berinhard committed:
Make sure delete operations work as expected for Polymorphic models

While testing the PR, I discovered this bug because I couldn't delete
applications I've created. After doing some research, I figured out this
is due to a known bug on django-polymorphic. More on this issue can be
found in this issue:

https://github.com/django-polymorphic/django-polymorphic/issues/229

berinhard committed:
Refactor how to reconfigure sponsor benefit to avoid delete operations

berinhard committed:
Make sure the assets are also being updated

berinhard committed:
Add docstring to make it more clear that the input assets are decoupled from their configuration

berinhard committed:
Fix bad context variable name

berinhard committed:
Fields that configure relationship between assets and configuration should be read only for editing
